index.html

```diff
@@ -12003,6 +12003,44 @@ <h2>Definitions of States and Properties (all aria-* attributes)</h2>
 				</tbody>
 			</table>
 		</div>
+		<div class="property" id="aria-labelsynonyms">
+			<pdef>aria-label</pdef>
+			<div class="property-description">
+				<p><a>Defines</a> an optional string value that provides alternate activation labels for the the current element, separated by commas. See related <pref>aria-label</pref> and <pref>aria-labelledby</pref>.</p>
+				<p>The purpose of <pref>aria-labelsynonyms</pref> is to allow a user of assistive technology such as voice activation software to activate controls through multiple names. Other uses may include "type-ahead" in a keyboard access utility. For example, a button that closes a dialog may be labeled "close" for a screen reader, but sighted user of voice activation may say "tap x" or "tap dismiss."</p>
+				<pre class="example highlight">
+&lt;button aria-label="close" <strong>aria-labelsynonyms="dismiss, clear, x"</strong>&gt;❎&lt;/button&gt;
+				</pre>
+				<p>Authors MUST NOT specify <code>aria-labelsynonyms</code> on an element that has no accessible name as determined by the <a href="#textalternativecomputation">accessible name and description computation</a>.</p>
```

Comment:
I'm not sure it's necessary. For authoring and implementation simplicity, I tend to avoid rule complexity unless it's vital to prevent a known anti-pattern. For example, imagine this is in a calendar table:

```html
…<td class="wed"><span aria-labelsynonyms="first day">1</span></td>…
```

A voice control user speaks "move to first day"… Why prevent that from working? Sure, we could call that an author error (should be on the `cell`, not on `generic`), but then it penalizes the author and user. As written, it's costless to other AT, and allows new speech or typeahead interactions to work.

Note: I'm sure we can come up with a better example. That was just the first I thought of.

Reply:
Considering removing the `Authors MUST NOT` statement above.

Reply:
if this can be implemented consistently in a way that ancestor interactive elements will be able to collect descendant label synonyms, then i can't disagree with your point.

So, if someone were to do
```
<button aria-labelsynonyms="first">
  <span aria-labelsynonyms="april fools">1</span>
</button>
```
And the button's name was "1" with the label synonyms of "first" and "april fools", then yeh.  that makes sense to me.

Reply:
@cookiecrook I'd like confirmation that this is costless to AT. It seems like it would require every element which could have label synonyms to walk through all its child elements to determine if there are any. While I appreciate wanting to be permissive this does seem like it would increase the implementation cost.

Reply:
Scott's example raises a good point about elements (like button) where child elements are considered presentational.

In either case, perf would depend on implementations… If it were always computed (to expose in the API tree), then you're probably right that a deep dive would have some cost. If it were only computed or collected once the user has asked to trigger it ("tap 'April first'") then perf cost would likely be minimal. I'll ask around and would like to hear @aleventhal's thoughts too.

Reply:
After some internal discussion, there is definitely the possibility of performance cost on Scott's example, because of the `chidrenPresentational` context with deeply nested sub-nodes of button.

```html
<button aria-labelsynonyms="first">
  <span>
    <span>
      <span>
        <span aria-labelsynonyms="april fools">1</span>
      </span>
    </span>
  </span>
</button>
```

The same is a not as concerning for my prior example, since the child nodes of `<td>` are not presentational, so the rendering engine would not need to deep dive to collate these. It vends exposes the label synonyms for the currently rendered node.

```html
<td class="wed">
  <span aria-labelsynonyms="first, first day, first wednesday, etc">1</span>
</td>
```

So I'm open to either limiting to name-able elements as @scottaohara suggested, or leaving global but disallowing in the context of presentational child nodes or other elements with role `presentation`/`none`.

Reply:
Resolved in a later commit